eth/downloader/peer.go

```diff
@@ -409,7 +419,12 @@ func (ps *peerSet) BodyIdlePeers() ([]*peer, int) {
 	idle := func(p *peer) bool {
 		return atomic.LoadInt32(&p.blockIdle) == 0
 	}
-	return ps.idlePeers(62, 64, idle)
+	throughput := func(p *peer) float64 {
```

Comment:
This `func` is essentially duplicated three times - could be simplified by instead passing a pointer to `idlePeers` - and do the locking there instead. Locking could also be avoided here if throughout values are made atomic.